### ISSUE TITLE
TYPO3 v9: Moving pages can lead to two entries

### ISSUE THREAD

**Issue by AMartinNo1** [wazum/sluggi#42]
Hi @wazum,

we've implemented a trailing slash for a customer and noticed that two records will be added to `sys_redirects`.  By looking into the code I have noticed that when a custom default `PageTypeSuffix` is configured two redirects will be added by the `DatamapHook`.

https://github.com/wazum/sluggi/blob/0846762ae7cde6f9274e5cf1b8cc4287264a97f1/Classes/Backend/Hook/DatamapHook.php#L322

```
            if ($currentSlug !== $slug) {
                // Remove old redirects matching the previous slug
                $this->deleteRedirect($siteHost, $sitePath . $currentSlug);
                $this->createRedirect($siteHost, $sitePath . $currentSlug, $pageId);
                if (!empty($variant)) {
                    $this->deleteRedirect($siteHost, $sitePath . $currentSlug . $variant);
                    $this->createRedirect($siteHost, $sitePath . $currentSlug . $variant, $pageId);
                }
            }
```

**Comment by wazum** [comment]
Hi @AMartinNo1 thanks for reporting, I'll take a look at it

**Comment by AMartinNo1** [comment]
In my opinion it should be btw. either of above `$this->createRedirect`-calls but not both. Hence, if `!empty($variant)` go with  `$variant`. Otherwise, well, don't.
I am not fully aware of the whole extension though. So, please take this as a hint but not necessarily solution 🙂

**Comment by moe2k** [comment]
Great work, the bug is successfully solved with 1.11.0. Tested with TYPO3 9.5.25.
